craft_parts/packages/deb.py

```diff
@@ -489,7 +490,12 @@ def unpack_stage_packages(
         cls, *, stage_packages_path: pathlib.Path, install_path: pathlib.Path
     ) -> None:
         """Unpack stage packages to install_path."""
-        for pkg_path in stage_packages_path.glob("*.deb"):
+        stage_packages = stage_packages_path.glob("*.deb")
+
+        if not stage_packages:
```

Comment:
`stage_packages` is a generator, it's `bool` will always be True. You need to consume it.

Reply:
Good catch! This line actually comes from snapcore/snapcraft/pull/3529 and it was added considering the behavior of `glob.glob()` but not `Path.glob()`. @sergiusens may want to fix it there as well.

Reply:
Fixed, and also opened a PR for snapcraft.